tests/infra/rates.py

```diff
@@ -2,84 +2,110 @@
-            for key in sorted(self.data.keys()):
-                file.write(key + " : " + str(self.data[key]))
-                file.write("\n")
-
-    def _get_hist(self):
-        rv = infra.proc.ccall(
-            "./client",
-            "--host={}".format(self.primary.host),
-            "--port={}".format(self.primary.tls_port),
-            "--ca=networkcert.pem",
-            "userrpc",
-            "--cert=user1_cert.pem",
-            "--pk=user1_privk.pem",
-            "--req=getTxHist.json",
-            log_output=False,
-        )
-
-        result = rv.stdout.decode().split("\n")[1]
-        result = json.loads(result)
-        histogram = result["result"]["tx_hist"]["histogram"]
-        LOG.info("Filtering histogram results...")
-        for key in histogram:
-            if histogram[key] > 0:
-                self.data[key] = histogram[key]
-        self.data["low"] = result["result"]["tx_hist"]["low"]
-        self.data["high"] = result["result"]["tx_hist"]["high"]
-        self.data["underflow"] = result["result"]["tx_hist"]["underflow"]
-        self.data["overflow"] = result["result"]["tx_hist"]["overflow"]
+        with open("tx_rates.json", "w") as mfile:
+            json.dump(self.all_metrics, mfile)
+
+    def _get_metrics(self):
+        with self.primary.user_client(format="json") as client:
+            rv = client.rpc("getMetrics", {})
+            result = rv.to_dict()
+            result = result["result"]["metrics"]
+            self.all_metrics = result
+
+            all_rates = []
+            all_durations = []
+            if "tx_rates" in result:
+                rates = result["tx_rates"]
+                if rates is None:
+                    LOG.info("No tx rate metrics found...")
+                else:
+                    for key in rates:
+                        all_rates.append(rates[key]["rate"])
+                        all_durations.append(float(rates[key]["duration"]))
+                    self.tx_rates_data = all_rates
+
+            else:
+                LOG.info("No tx rate metrics found...")
+
+            if "histogram" not in result:
+                LOG.info("No histogram metrics found...")
+            else:
+                histogram = result["histogram"]
+                histogram_buckets = histogram["buckets"]
+
+                LOG.info("Filtering histogram results...")
+                hist_data = {}
+
+                for key in histogram_buckets:
+                    if histogram_buckets[key] > 0:
+                        range_1, range_2 = key.split("..")
+                        hist_data[int(range_1)] = (range_2, histogram_buckets[key])
+
+                ordered_data = collections.OrderedDict(
+                    sorted(hist_data.items(), key=lambda x: x[0])
+                )
+                self.histogram_data["histogram"] = {}
+                buckets = []
+                rates = []
+                for key, value_tuple in ordered_data.items():
```

Comment:
No need for an OrderedDict, since you're going to bind the pairs, you can just pass the list of pairs directly here.

Reply:
I use an ordered dictionary so that when the histogram is displayed, the buckets are displayed in order